lib/completions/endpoints/vllm.rb

```diff
@@ -44,16 +30,13 @@ def provider_id
         private
 
         def model_uri
-          if llm_model&.url && !llm_model&.url == LlmModel::RESERVED_VLLM_SRV_URL
-            return URI(llm_model.url)
-          end
-
-          service = DiscourseAi::Utils::DnsSrv.lookup(SiteSetting.ai_vllm_endpoint_srv)
-          if service.present?
+          if llm_model.url == LlmModel::RESERVED_VLLM_SRV_URL
+            service = DiscourseAi::Utils::DnsSrv.lookup(SiteSetting.ai_vllm_endpoint_srv)
             api_endpoint = "https://#{service.target}:#{service.port}/v1/chat/completions"
           else
-            api_endpoint = "#{SiteSetting.ai_vllm_endpoint}/v1/chat/completions"
+            api_endpoint = llm_model.url
           end
+
           @uri ||= URI(api_endpoint)
         end
```

Comment:
Didn't we talk about using special `srv://` detection instead of ENV?

Like 

```ruby
        def model_uri
          @uri ||=
            if !llm_model.url.starts_with?("srv://")
              URI(llm_model.url)
            else
              service = DiscourseAi::Utils::DnsSrv.lookup(llm_model.url)
              URI("https://#{service.target}:#{service.port}/v1/chat/completions")
            end
        end
```

Reply:
I forgot about that, sorry. Will change